cdc/capture/capture.go

```diff
@@ -600,7 +619,17 @@ func (c *captureImpl) Drain(ctx context.Context) <-chan struct{} {
 	const drainInterval = 100 * time.Millisecond
 	done := make(chan struct{})
 	go func() {
-		defer close(done)
+		defer func() {
+			lease := c.session.Lease()
+			_, err := c.EtcdClient.Client.Revoke(ctx, lease)
```

Comment:
Why not revoke lease in `capture.resign()`?

Reply:
I  think `capture.resign` is called, if the capture win the `campaign`.

For capture which is not the owner, `resign` won't be called.